lldb/test/API/functionalities/data-formatter/type-recognizer/value-object-cast/TestRecognizedValue.py

```diff
@@ -0,0 +1,514 @@
+"""
+Use lldb Python API to test ValueObject type recognizer
+"""
+
+
+import lldb
+from lldbsuite.test.decorators import *
+from lldbsuite.test.lldbtest import *
+from lldbsuite.test import lldbutil
+
+
+class RecognizedValueTestCase(TestBase):
+    def setUp(self):
+        # Call super's setUp().
+        TestBase.setUp(self)
+
+        # Execute the cleanup function during test case tear down.
+        self.addTearDownHook(self.cleanup)
+
+        # Find the line number to break for main.cpp.
+        self.br_line_cast_func = line_number(
+            "main.cpp", "// Break here in cast_func()."
+        )
+
+        # Prepare executable.
+        self.build()
+        self.exe = self.getBuildArtifact("a.out")
+
+    def test_find_variable(self):
+        """Test recognizing variables using the FindVariable."""
+        # Create a target from the debugger.
+        target = self.dbg.CreateTarget(self.exe)
+        self.assertTrue(target, VALID_TARGET)
+
+        # Set up our breakpoints:
+        br_cast_func = target.BreakpointCreateByLocation(
+            "main.cpp", self.br_line_cast_func
```

Comment:
I'd like to see call to `line_number` inlined here, so it's easier to see what are we breaking on